src/main/java/seedu/address/model/person/Person.java

```diff
@@ -26,7 +26,7 @@ public class Person {
     // Data fields
     private final Role role;
     private final Set<Skill> skills = new HashSet<>();
-    private final Optional<String> match;
+    private Optional<String> match;
```

Comment:
Indeed, this can be a good discussion topic.

Unlike Job which works with a HashSet and hence it can be modified, it has to do with how as an Optional<String>, I cannot convert it to an Optional.empty() which is an entirely different reference. 

I thought about this quite long, and the solutions I came up with was to either do Optional.of("") and do empty string checks which might break many things, create entirely new objects and set them within the list (similar to match), or simply remove the final and directly mutate the object.